content/post/000-ipfs-in-web-browsers.md

```diff
@@ -0,0 +1,74 @@
+
+Decentralization feels slow in providing user-centric approaches to the internet's biggest problems. Right now true user agency is often too technical, too fragile and too time-consuming to be the default choice.
+
+But we're making headway. And we'd like to share some collaborations the IPFS project has had in the works for a while, which bring us just a few steps closer to making unmediated access to information *just work*... by solving that "last mile" problem and *integrating IPFS directly into web browsers*.
+
+## Stages
+
+The path to a truly decentralized web is a long one. For over 30 years the browser has been a *client* - but a foundational concept in P2P systems is that a participant is both a client *and a server*. Web browser vendors and web standards bodies have not designed for this architectural shift, so we're breaking it down into steps.
+
+Initially we're aiming to ease access to the IPFS HTTP gateways - creating a bridge between the billions of people on the centralized web and content on the distributed web. We expect some experimentation in the next phase - adjusting node behavior based on browser APIs, device capabilities, bandwidth availability and power consumption - growing adoption with our browser partners as we learn, and building common understanding at standards bodies. Finally we aim to have multiple browsers implementing IPFS natively and interoperably.
+
+(INSERT STAGES DIAGRAM)
+
+## Firefox
+
+> Perhaps it makes sense to mention Firefox as a full recap-like section here, afaik we did not blog about libdweb at all yet. 
```

Comment:
+1! I think it's a great idea. In the spirit of trail-blazing and documenting where browser collaborations began, this makes a lot of sense. Ex "as of Firefox 59, the browser has had support for ipfs:// via decentralized protocol handlers."

Reply:
Flashback, I implemented that tab hiding API...